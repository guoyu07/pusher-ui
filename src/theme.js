const fontFamilies = [
  '-apple-system',
  'BlinkMacSystemFont',
  'Segoe UI',
  'Roboto',
  'Oxygen',
  'Ubuntu',
  'Cantarell',
  'Fira Sans',
  'Droid Sans',
  'Helvetica Neue',
  'sans-serif',
].map(name => `"${name}"`).join(', ');


export const fontFamily = fontFamilies;
export const white = '#fff';
export const black = '#1e1e1e';
export const grey = '#e7ebf3';
export const darkGrey = '#d6d6d6';
export const darkDarkGrey = '#989898';
export const lightGrey = '#e7ebf3';
export const primary = '#549eff';
export const success = '#319a31';
export const warning = '#ffaf1c';
export const danger = '#ca0808';
export const zIndex1 = 1;
export const zIndex2 = 2;
export default {
  fontFamily,
  white,
  black,
  grey,
  darkGrey,
  darkDarkGrey,
  lightGrey,
  primary,
  success,
  warning,
  danger,
  zIndex1,
  zIndex2,
};
