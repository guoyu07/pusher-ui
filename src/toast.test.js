import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'glamorous';

import { Toast } from './index';
import * as theme from './theme';

describe('Toast', () => {
  const defaultProps = {
    text: 'I am a toast!',
  };
  const createSnapshot = (props = defaultProps) => {
    expect(
      renderer
        .create(
          <ThemeProvider theme={theme}>
            <Toast {...props} />
          </ThemeProvider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  };

  it('should render a toast', () => {
    createSnapshot();
  });

  it('should render take a text prop', () => {
    createSnapshot({
      ...defaultProps,
      text: 'Some message to the user',
    });
  });
});
