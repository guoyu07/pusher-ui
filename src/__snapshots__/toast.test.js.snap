// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should render a toast 1`] = `
.glamor-0,
[data-glamor-0] {
  max-width: 22.5rem;
  margin: 0.375rem;
  padding-top: 0.75rem;
  padding-right: 1.125rem;
  padding-bottom: 0.75rem;
  padding-left: 1.125rem;
  pointer-events: auto;
  border-radius: 2px;
  font-weight: 500;
  cursor: pointer;
  box-shadow: 0 1px 4px 1px rgba(50, 52, 58, 0.13);
  color: #2b303b;
  background-color: #fff;
}

<div
  className="glamor-0"
  onClick={[Function]}
  style={
    Object {
      "WebkitTransform": "translate3d(100%, 0, 0)",
      "opacity": 0,
      "transform": "translate3d(100%, 0, 0)",
    }
  }
>
  I am a toast!
</div>
`;

exports[`Toast should render take a text prop 1`] = `
.glamor-0,
[data-glamor-0] {
  max-width: 22.5rem;
  margin: 0.375rem;
  padding-top: 0.75rem;
  padding-right: 1.125rem;
  padding-bottom: 0.75rem;
  padding-left: 1.125rem;
  pointer-events: auto;
  border-radius: 2px;
  font-weight: 500;
  cursor: pointer;
  box-shadow: 0 1px 4px 1px rgba(50, 52, 58, 0.13);
  color: #2b303b;
  background-color: #fff;
}

<div
  className="glamor-0"
  onClick={[Function]}
  style={
    Object {
      "WebkitTransform": "translate3d(100%, 0, 0)",
      "opacity": 0,
      "transform": "translate3d(100%, 0, 0)",
    }
  }
>
  Some message to the user
</div>
`;
